Remove fbgemm_is_cpu_supported in favor of torch.backends.quantized.supported_qengines

dzhulgakov opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26897 Fix misuages for TORCH_CHECK/TORCH_INTERNAL_ASSERT with string
* **#26840 Remove fbgemm_is_cpu_supported in favor of torch.backends.quantized.supported_qengines**

Cleaning up top-level namespace. Also cosmetic changes to torch.backends.quantized

Also fixes QNNPACK routing conditions which got surfaced once I adjusted the default engine selection

Differential Revision: [D17604403](https://our.internmc.facebook.com/intern/diff/D17604403)

Commit:
Remove fbgemm_is_cpu_supported in favor of torch.backends.quantized.supported_qengines

[ghstack-poisoned]

Commit:
Update on "Remove fbgemm_is_cpu_supported in favor of torch.backends.quantized.supported_qengines"


Cleaning up top-level namespace. Also cosmetic changes to torch.backends.quantized


[ghstack-poisoned]

Commit:
Update on "Remove fbgemm_is_cpu_supported in favor of torch.backends.quantized.supported_qengines"


Cleaning up top-level namespace. Also cosmetic changes to torch.backends.quantized


[ghstack-poisoned]

supriyar reviewed (approved):
LGTM, Thanks!

Commit:
Update on "Remove fbgemm_is_cpu_supported in favor of torch.backends.quantized.supported_qengines"


Cleaning up top-level namespace. Also cosmetic changes to torch.backends.quantized


[ghstack-poisoned]

jerryzh168 commented on test/test_torch.py:
```diff
@@ -2070,8 +2070,8 @@ def test_inference(default_dtype):
-    def test_qengnie(self):
-        qengines = torch.backends.quantized.get_supported_qengines()
+    def test_qengine(self):
+        qengines = torch.backends.quantized.supported_engines
```
supported_qengines?

dzhulgakov replied:
there's already "quantized" in the name, so I simplified the name

jerryzh168 replied:
we have `supportedQEngine` in c++, I think it's better to align the names.. maybe change both of them either to `qengine` or `engine`, I think supported_qengine is better since `QEngine` is one word.. just like `QScheme`

Commit:
Update on "Remove fbgemm_is_cpu_supported in favor of torch.backends.quantized.supported_qengines"


Cleaning up top-level namespace. Also cosmetic changes to torch.backends.quantized

Differential Revision: [D17604403](https://our.internmc.facebook.com/intern/diff/D17604403)

[ghstack-poisoned]

Commit:
Update on "Remove fbgemm_is_cpu_supported in favor of torch.backends.quantized.supported_qengines"


Cleaning up top-level namespace. Also cosmetic changes to torch.backends.quantized

Differential Revision: [D17604403](https://our.internmc.facebook.com/intern/diff/D17604403)

[ghstack-poisoned]

Commit:
Update on "Remove fbgemm_is_cpu_supported in favor of torch.backends.quantized.supported_qengines"


Cleaning up top-level namespace. Also cosmetic changes to torch.backends.quantized

Also fixes QNNPACK routing conditions which got surfaced once I adjusted the default engine selection

Differential Revision: [D17604403](https://our.internmc.facebook.com/intern/diff/D17604403)

[ghstack-poisoned]

Commit:
Update on "Remove fbgemm_is_cpu_supported in favor of torch.backends.quantized.supported_qengines"


Cleaning up top-level namespace. Also cosmetic changes to torch.backends.quantized

Also fixes QNNPACK routing conditions which got surfaced once I adjusted the default engine selection

Differential Revision: [D17604403](https://our.internmc.facebook.com/intern/diff/D17604403)

[ghstack-poisoned]
